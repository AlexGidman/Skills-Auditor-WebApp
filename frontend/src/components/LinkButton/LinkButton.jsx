import React from "react";
import { Link } from "react-router-dom";
import cx from "classnames";

import styles from "./LinkButton.module.css";

export const LinkButton = ({ path = "/", name = "Link", className, disabled = false, ...rest }) => {
    return (
        <Link
            className={cx(styles.component, className, disabled && styles.disabled)}
            to={path}
            {...rest}
        >
            {name}
        </Link>
    );
};
