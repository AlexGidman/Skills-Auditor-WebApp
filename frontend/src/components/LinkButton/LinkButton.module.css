@import "../Button/Button.module.css";

.component {
    composes: component from "../Button/Button.module.css";
}

.component:link,
.component:visited {
    text-decoration: none;
}

.component:hover,
.component:active,
.component:focus-within {
    text-decoration: none;
}

.disabled {
    pointer-events: none;
    cursor: default;
    color: black !important;
    background-color: var(--c-grey) !important;
}
