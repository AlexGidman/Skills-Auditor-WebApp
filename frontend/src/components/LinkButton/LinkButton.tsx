import React from "react";
import { Link } from "react-router-dom";
import cx from "classnames";

import styles from "./LinkButton.module.css";

interface Props
    extends React.ComponentProps<"a">,
        Pick<React.ComponentProps<"button">, "disabled"> {
    path?: string;
    name?: string;
}

export const LinkButton = ({
    path = "/",
    name = "Link",
    className,
    disabled = false,
}: Props): React.ReactElement => {
    return (
        <Link className={cx(styles.component, className, disabled && styles.disabled)} to={path}>
            {name}
        </Link>
    );
};
