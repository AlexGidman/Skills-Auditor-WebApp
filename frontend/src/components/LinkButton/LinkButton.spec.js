import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { LinkButton } from "./LinkButton";

describe("LinkButton", () => {
    it("renders correctly when passed props", async () => {
        const { container } = render(
            <MemoryRouter>
                <LinkButton path="/mypath" name="My Path" />
            </MemoryRouter>,
        );

        const link = await screen.findByRole("link", { name: "My Path" });
        expect(link).toBeInTheDocument();
        expect(link).toHaveAttribute("href", "/mypath");
        expect(container).toMatchSnapshot();
    });
    it("renders correctly when not passed props", async () => {
        const { container } = render(
            <MemoryRouter>
                <LinkButton />
            </MemoryRouter>,
        );

        const link = await screen.findByRole("link", { name: "Link" });
        expect(link).toBeInTheDocument();
        expect(link).toHaveAttribute("href", "/");
        expect(container).toMatchSnapshot();
    });
});
