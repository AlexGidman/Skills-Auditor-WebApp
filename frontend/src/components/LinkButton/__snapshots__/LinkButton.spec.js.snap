// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders correctly when not passed props 1`] = `
<div>
  <a
    class="component"
    href="/"
  >
    Link
  </a>
</div>
`;

exports[`LinkButton renders correctly when passed props 1`] = `
<div>
  <a
    class="component"
    href="/mypath"
  >
    My Path
  </a>
</div>
`;
